src/main/scala/lms/core/codegen/scala_codegen.scala

```diff
@@ -186,14 +186,15 @@ class CompactScalaCodeGen extends CompactTraverser {
     }
     withWraper(wraper _)(f)
   }
-  def quoteBlock(b: Block): Unit = {
+  def quoteBlock(b: Block, argType: Boolean = false): Unit = {
     def eff = quoteEff(b.ein)
-    if (b.in.length == 0) {
-      quoteBlock(traverse(b)) //FIXME(GW): discard eff if arg length == 0?
-    } else {
-      val argsStr = b.in.map(arg => quote(arg)).mkString("(", ", ", s")$eff => ")
-      quoteBlock(argsStr)(traverse(b))
-    }
+    val args = b.in.map({ arg =>
+      val tp = if (argType) (": Int") else "" //FIXME(GW): get the type annotation, add a typeMap?
+      quote(arg) + tp
+    })
+    val argsStr = if (args.length == 1 && !argType) s"${args.head}$eff => "
+                  else args.mkString("(", ", ", s")$eff => ")
+    quoteBlock(argsStr)(traverse(b))
```

Comment:
@Kraks please check if this change breaks any of your features.

Reply:
It looks good to me, I will have a test on my code and come back later.